### PR TITLE
Modify the data structure to store telemetry metrics

### PR DESCRIPTION
**Pull request by lkomali**
Earlier, the telemetry metrics are stored as a list of lists, where each inner list corresponds to metric value of different GPUs for current measurement.
This data structure inflexible and provided challenges in utilizing existing methods in statistics.py , which led to duplication.
This PR  uses a different data structure.
With this change, the metrics are now stored as
'gpu_power_usage': {
        'gpu0': [27.01]
    },
    'gpu_utilization': {
        'gpu0': [75.5]
    },
    'energy_consumption': {
        'gpu0': [123.56]
    }
 With this new data structure, it's easy to leverage the methods in statistics.py

### CODE REVIEW

File: genai-perf/genai_perf/metrics/telemetry_metrics.py
Comment (nv-hwoo) [inline genai-perf/genai_perf/metrics/telemetry_metrics.py:82]: if you set `defaultdict(list)` to store the metrics, then I think you can simply do 
```suggestion
                    getattr(self, metric_key)[gpu_name].extend(values)
```

Comment (lkomali) [reply genai-perf/genai_perf/metrics/telemetry_metrics.py]: I modified.

File: genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py
Comment (debermudez) [inline genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py:70]: This is sample data?

File: genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py
Comment (debermudez) [inline genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py:93]: why was this check added?
does the metrics_data include comments?

Comment (lkomali) [reply genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py:93]: Yep, it has comments.
Below is the a part of metrics from Triton 
```
# HELP nv_inference_request_success Number of successful inference requests, all batch sizes
# TYPE nv_inference_request_success counter
nv_inference_request_success{model="vllm_model",version="1"} 7918
# HELP nv_inference_request_failure Number of failed inference requests, all batch sizes
# TYPE nv_inference_request_failure counter
nv_inference_request_failure{model="vllm_model",reason="BACKEND",version="1"} 0
nv_inference_request_failure{model="vllm_model",reason="CANCELED",version="1"} 0
nv_inference_request_failure{model="vllm_model",reason="OTHER",version="1"} 0
nv_inference_request_failure{model="vllm_model",reason="REJECTED",version="1"} 0
# HELP nv_inference_count Number of inferences performed (does not include cached requests)
# TYPE nv_inference_count counter
nv_inference_count{model="vllm_model",version="1"} 7918
# HELP nv_inference_exec_count Number of model executions performed (does not include cached requests)
# TYPE nv_inference_exec_count counter
nv_inference_exec_count{model="vllm_model",version="1"} 7918
# HELP nv_inference_request_duration_us Cumulative inference request duration in microseconds (includes cached requests)
# TYPE nv_inference_request_duration_us counter
nv_inference_request_duration_us{model="vllm_model",version="1"} 452661337
# HELP nv_inference_queue_duration_us Cumulative inference queuing duration in microseconds (includes cached requests)
# TYPE nv_inference_queue_duration_us counter
nv_inference_queue_duration_us{model="vllm_model",version="1"} 128932
# HELP nv_inference_compute_input_duration_us Cumulative compute input duration in microseconds (does not include cached requests)
# TYPE nv_inference_compute_input_duration_us counter
nv_inference_compute_input_duration_us{model="vllm_model",version="1"} 192290
# HELP nv_inference_compute_infer_duration_us Cumulative compute inference duration in microseconds (does not include cached requests)
# TYPE nv_inference_compute_infer_duration_us counter
nv_inference_compute_infer_duration_us{model="vllm_model",version="1"} 452294570
# HELP nv_inference_compute_output_duration_us Cumulative inference compute output duration in microseconds (does not include cached requests)
```

File: genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py
Comment (lkomali) [reply genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py:70]: That's the sample how the method stores the data.

**Comment by debermudez** [comment]
I will wait for @nv-hwoo to approve because he has outstanding comments.
I think this is good to go.

**Review by nv-hwoo — DISMISSED** [review]
oops sorry, didn't realize you were waiting on my approval! lgtm 👍

**Review by nv-hwoo — APPROVED** [review]

**Pull request merged**
